react/src/components/SessionUsageMonitor.tsx

```diff
@@ -0,0 +1,254 @@
+import { bytesToGB, toFixedFloorWithoutTrailingZeros } from '../helper';
+import Flex from './Flex';
+import { ResourceTypeIcon } from './ResourceNumber';
+import { SessionUsageMonitorFragment$key } from './__generated__/SessionUsageMonitorFragment.graphql';
+import { Col, Progress, ProgressProps, Row, Typography, theme } from 'antd';
+import { createStyles } from 'antd-style';
+import graphql from 'babel-plugin-relay/macro';
+import _ from 'lodash';
+import { useMemo } from 'react';
+import { useFragment } from 'react-relay';
+
+const useStyles = createStyles(({ css, token }) => ({
+  progress: css`
+    .ant-progress-text {
+      font-size: ${token.fontSizeSM} !important;
+    }
+    .ant-progress-inner {
+    }
+    .ant-typography {
+      color: ${token.colorTextSecondary} !important;
+      font-size: ${token.fontSizeSM}px !important;
+    }
+  `,
+}));
+
+interface SessionUsageMonitorProps extends ProgressProps {
+  sessionFrgmt: SessionUsageMonitorFragment$key | null;
+  col?: 1 | 2 | 3 | 4;
+  size?: ProgressProps['size'];
+}
+
+// compute session node query returns accelerator type as `[device_name]_[util|mem]`
+// so we need to split via '_' and get the first part to map to the resource type in `resourceTypeIconSrcMap`
+type acceleratorType =
```

Comment:
this is the same code with `knownAcceleratorResourceSlotNames` of hooks/backendai.tsx